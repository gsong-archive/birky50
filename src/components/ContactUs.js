import React from "react";

import { Row, Col, Button, Form, FormGroup, Label, Input } from "reactstrap";

export default () => {
  const labelWidth = 1;
  const inputWidth = 5;
  const sendTo = "birky50@damacy.net";

  return (
    <Row>
      <Col>
        <h1>🙋🏻‍♀️ Questions?</h1>
        <Form action={`https://formspree.io/${sendTo}`} method="POST">
          <FormGroup row>
            <Label for="contact_us_name" md={labelWidth}>
              Name
            </Label>
            <Col lg={inputWidth} md={inputWidth + 1}>
              <Input
                type="text"
                name="name"
                id="contact_us_name"
                placeholder="Your Name"
                required
              />
            </Col>
          </FormGroup>
          <FormGroup row>
            <Label for="contact_us_email" md={labelWidth}>
              Email
            </Label>
            <Col lg={inputWidth} md={inputWidth + 1}>
              <Input
                type="email"
                name="email"
                id="contact_us_email"
                placeholder="Your Email"
                required
              />
            </Col>
          </FormGroup>
          <FormGroup row>
            <Label for="contact_us_message" md={labelWidth}>
              Message
            </Label>
            <Col lg={inputWidth} md={inputWidth + 1}>
              <Input
                type="textarea"
                name="message"
                id="contact_us_message"
                placeholder="Message"
                rows="3"
                required
              />
            </Col>
          </FormGroup>
          <input
            type="hidden"
            name="_subject"
            value="Birky 50th Anniversary Message"
          />
          <input type="text" name="_gotcha" style={{ display: "none" }} />
          <Row>
            <Col md={{ size: inputWidth, offset: labelWidth }}>
              <Button type="submit" color="primary">
                Send
              </Button>
            </Col>
          </Row>
        </Form>
      </Col>
    </Row>
  );
};
