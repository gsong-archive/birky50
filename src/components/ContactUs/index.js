//@flow strict
import React from "react";

import facepaint from "facepaint";
import styled, { css } from "react-emotion";

import SectionHeader from "../shared/SectionHeader";
import { btn, formInput } from "../../styles";
import { colors } from "../../styles/variables";
import { convertListToEm } from "../utils";
import { supportsGrid } from "../../styles/cssFeatures";

import type { Props } from "../Section.type";

export default ({ LabelComponent, sectionLabel }: Props) => {
  const sendTo = "birky50@damacy.net";

  return (
    <React.Fragment>
      <SectionHeader>
        <LabelComponent label={sectionLabel} />
      </SectionHeader>

      <Form action={`https://formspree.io/${sendTo}`} method="POST">
        <Label htmlFor="contact_us_name">Name</Label>
        <Input
          type="text"
          name="name"
          id="contact_us_name"
          placeholder="Your Name"
          required
        />

        <Label htmlFor="contact_us_email">Email</Label>
        <Input
          type="email"
          name="email"
          id="contact_us_email"
          placeholder="Your Email"
          required
        />

        <Label htmlFor="contact_us_message">Message</Label>
        <Textarea
          name="message"
          id="contact_us_message"
          placeholder="Message"
          rows="3"
          required
        />

        <input
          type="hidden"
          name="_subject"
          value="Birky 50th Anniversary Message"
        />
        <input type="hidden" name="_gotcha" />

        <Button type="submit">Send</Button>
      </Form>
    </React.Fragment>
  );
};

const breakpoints = convertListToEm([600]);
const mq = facepaint(breakpoints.map(bp => `@media (min-width: ${bp}em)`));

const IEMarginTop = css(!supportsGrid && mq({ marginTop: [null, "1rem"] }));

const Form = styled.form(
  mq({
    display: [null, "grid"],
    gridTemplateColumns: [null, "5rem 30rem"],
    gridRowGap: [null, "1rem"],
  }),
  !supportsGrid &&
    mq({
      display: [null, "flex"],
      flexWrap: [null, "wrap"],
      alignItems: [null, "baseline"],
      width: [null, "40rem"],
    })
);

const Label = styled.label(
  mq({ marginTop: [null, "0.575rem"] }),
  !supportsGrid &&
    mq({
      flex: [null, "0 0 5rem"],
    })
);

const Input = styled.input(
  formInput,
  { display: "block" },
  mq({
    width: ["100%", "auto"],
    margin: [".5rem 0 1rem", 0],
  }),
  !supportsGrid &&
    mq({
      flex: [null, "0 0 30rem"],
    }),
  IEMarginTop
);

const Textarea = Input.withComponent("textarea");

const Button = styled.button(
  btn,
  css`
    grid-column-start: 2;
    justify-self: start;
    cursor: pointer;
    background-color: ${colors.primary};
    color: white;
  `,
  !supportsGrid && mq({ marginLeft: [null, "5rem"] }),
  IEMarginTop
);
