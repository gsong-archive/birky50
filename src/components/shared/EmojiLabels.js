import React from "react";

const EmojiLabel = ({ emoji, ariaLabel, label }) => (
  <React.Fragment>
    <span role="img" aria-label={ariaLabel}>
      {emoji}
    </span>{" "}
    {label}
  </React.Fragment>
);

export const Airport = ({ label }) => (
  <EmojiLabel emoji="✈️" aria-label="airplane" label={label} />
);

export const Burger = ({ label }) => (
  <EmojiLabel emoji="🍔" aria-label="burger" label={label} />
);

export const Hotel = ({ label }) => (
  <EmojiLabel emoji="🏩" aria-label="hotel" label={label} />
);

export const PartyHat = ({ label }) => (
  <EmojiLabel emoji="🎉" aria-label="party hat" label={label} />
);

export const WomanBiking = ({ label }) => (
  <EmojiLabel emoji="🚴🏽‍♀️" aria-label="biking woman" label={label} />
);

export const WomanRaisingHand = ({ label }) => (
  <EmojiLabel emoji="🙋🏻‍♀️" aria-label="woman raising hand" label={label} />
);
