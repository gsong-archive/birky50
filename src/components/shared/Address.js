//@flow strict
import React from "react";

import { Link } from "../../styles/components";
import { Location } from "./EmojiLabels";

type Props = {
  address: string,
  url: string,
};

export default ({ address, url }: Props) => (
  <div>
    <Location label={<Link href={url}>{address}</Link>} />
  </div>
);
