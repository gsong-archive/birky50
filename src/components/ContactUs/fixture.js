//@flow
import { createFixture } from "react-cosmos-flow/fixture";

import ContactUs from "./index";
import { WomanRaisingHand } from "../shared/EmojiLabels";

export default createFixture({
  component: ContactUs,
  name: "Default",
  description: "`LabelComponent` prop is a React component.",
  props: { LabelComponent: WomanRaisingHand, sectionLabel: "Questions?" },
});
