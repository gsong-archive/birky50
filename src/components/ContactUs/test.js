//@flow
import React from "react";

import { render } from "react-testing-library";

import { mockSupportsGrid } from "testUtils/mocks";

import { WomanRaisingHand } from "../shared/EmojiLabels";

mockSupportsGrid(true);

const ContactUs = require("./index").default;

test("ContactUs form elements are properly labeled", () => {
  const NAME = "name";
  const EMAIL = "email";
  const MESSAGE = "message";
  const { getByLabelText } = render(
    <ContactUs LabelComponent={WomanRaisingHand} sectionLabel="Questions?" />
  );
  const nameInput = getByLabelText(RegExp(NAME, "i"));
  const emailInput = getByLabelText(RegExp(EMAIL, "i"));
  const messageInput = getByLabelText(RegExp(MESSAGE, "i"));

  // $FlowFixMe
  expect(nameInput.name).toBe(NAME);
  // $FlowFixMe
  expect(emailInput.name).toBe(EMAIL);
  // $FlowFixMe
  expect(messageInput.name).toBe(MESSAGE);
});

test("ContactUs renders properly", () => {
  const { container } = render(
    <ContactUs LabelComponent={WomanRaisingHand} sectionLabel="Questions?" />
  );

  expect(container).toMatchSnapshot();
});
