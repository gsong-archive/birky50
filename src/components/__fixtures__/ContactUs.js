import ContactUs from "../ContactUs";
import { WomanRaisingHand } from "../shared/EmojiLabels";

export default {
  component: ContactUs,
  props: { LabelComponent: WomanRaisingHand, sectionLabel: "Questions?" },
};
