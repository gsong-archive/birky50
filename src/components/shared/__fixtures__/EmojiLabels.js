//@flow
import React from "react";

import { createFixture } from "react-cosmos-flow/fixture";

import { generate } from "testUtils";

import { Airplane, EmojiLabel } from "../EmojiLabels";
import { Link } from "../../../styles/components";

export default [
  createFixture({
    name: "Predefined EmojiLabel with text label",
    component: Airplane,
    props: { label: generate.words() },
  }),
  createFixture({
    name: "Predefined EmojiLabel with link label",
    description: "`label` prop is a React node.",
    component: Airplane,
    props: {
      label: <Link href={generate.url()}>{generate.words()}</Link>,
    },
  }),
  createFixture({
    name: "With text label",
    component: EmojiLabel,
    props: {
      emoji: "ðŸ˜œ",
      ariaLabel: "stuck out tongue winking eye",
      label: generate.words(),
    },
  }),
  createFixture({
    name: "With link label",
    description: "`label` prop is a React node.",
    component: EmojiLabel,
    props: {
      emoji: "ðŸ˜œ",
      ariaLabel: "stuck out tongue winking eye",
      label: <Link href={generate.url()}>{generate.words()}</Link>,
    },
  }),
];
