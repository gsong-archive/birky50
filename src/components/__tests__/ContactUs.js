import React from "react";

import { render } from "react-testing-library";

import { mockSupportsGrid } from "testUtils/mocks";

import ContactUs from "../ContactUs";
import { WomanRaisingHand } from "../shared/EmojiLabels";

test("ContactUs form elements are properly labeled", () => {
  const NAME = "name";
  const EMAIL = "email";
  const MESSAGE = "message";
  const { getByLabelText } = render(
    <ContactUs LabelComponent={WomanRaisingHand} sectionLabel="Questions?" />
  );
  const nameInput = getByLabelText(RegExp(NAME, "i"));
  const emailInput = getByLabelText(RegExp(EMAIL, "i"));
  const messageInput = getByLabelText(RegExp(MESSAGE, "i"));

  expect(nameInput.name).toBe(NAME);
  expect(emailInput.name).toBe(EMAIL);
  expect(messageInput.name).toBe(MESSAGE);
});

test("ContactUs renders properly", () => {
  mockSupportsGrid(true);
  const { container } = render(
    <ContactUs LabelComponent={WomanRaisingHand} sectionLabel="Questions?" />
  );

  expect(container).toMatchSnapshot();
});
