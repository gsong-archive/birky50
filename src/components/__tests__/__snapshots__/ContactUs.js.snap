// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs renders properly 1`] = `
.emotion-1 {
  line-height: 1.1;
  margin-bottom: 1rem;
}

@media (max-width:350px) {
  .emotion-1 {
    font-size: 2.4rem;
  }
}

.emotion-0 {
  display: inline-block;
  margin-left: 1.4em;
  text-indent: -1.4em;
}

@media (min-width:600px) {
  .emotion-17 {
    display: grid;
    grid-template-columns: 5rem 30rem;
    grid-row-gap: 1rem;
  }
}

@media (min-width:600px) {
  .emotion-3 {
    margin-top: 0.575rem;
  }
}

.emotion-5 {
  border-radius: 0.25rem;
  font-size: 1rem;
  line-height: 1.5;
  padding: 0.375rem 0.75rem;
  border: 1px solid #ced4da;
  display: block;
  width: 100%;
  margin: .5rem 0 1rem;
}

@media (min-width:600px) {
  .emotion-5 {
    width: auto;
    margin: 0;
  }
}

.emotion-13 {
  border-radius: 0.25rem;
  font-size: 1rem;
  line-height: 1.5;
  padding: 0.375rem 0.75rem;
  border: 1px solid #ced4da;
  display: block;
  width: 100%;
  margin: .5rem 0 1rem;
}

@media (min-width:600px) {
  .emotion-13 {
    width: auto;
    margin: 0;
  }
}

.emotion-15 {
  border-radius: 0.25rem;
  display: inline-block;
  font-size: 1rem;
  text-align: center;
  vertical-align: middle;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.5rem 0.75rem;
  grid-column-start: 2;
  justify-self: start;
  cursor: pointer;
  background-color: #100d44;
  color: white;
}

<div>
  <h1
    id="uniqueId-description"
    class="emotion-1 emotion-2"
  >
    <span
      class="emotion-0"
    >
      <span
        role="img"
        aria-label="woman raising hand"
      >
        üôãüèª‚Äç‚ôÄÔ∏è
      </span>
      Questions?
    </span>
  </h1>
  <form
    action="https://formspree.io/birky50@damacy.net"
    method="POST"
    class="emotion-17 emotion-18"
  >
    <label
      for="contact_us_name"
      class="emotion-3 emotion-4"
    >
      Name
    </label>
    <input
      type="text"
      name="name"
      id="contact_us_name"
      placeholder="Your Name"
      required=""
      class="emotion-5 emotion-6"
    />
    <label
      for="contact_us_email"
      class="emotion-3 emotion-4"
    >
      Email
    </label>
    <input
      type="email"
      name="email"
      id="contact_us_email"
      placeholder="Your Email"
      required=""
      class="emotion-5 emotion-6"
    />
    <label
      for="contact_us_message"
      class="emotion-3 emotion-4"
    >
      Message
    </label>
    <textarea
      name="message"
      id="contact_us_message"
      placeholder="Message"
      rows="3"
      required=""
      class="emotion-13 emotion-14"
    />
    <input
      type="hidden"
      name="_subject"
      value="Birky 50th Anniversary Message"
    />
    <input
      type="hidden"
      name="_gotcha"
    />
    <button
      type="submit"
      class="emotion-15 emotion-16"
    >
      Send
    </button>
  </form>
</div>
`;
